src/arrays/programs_involving_data_sequences/medium/q7/MainTest.java

```diff
@@ -0,0 +1,61 @@
+package arrays.programs_involving_data_sequences.medium.q7;
+
+import global.BaseTest;
+import global.MethodTest;
+import global.tools.CustomAssertions;
+import global.utils.ArrayUtil;
+import global.variables.Clause;
+import global.variables.clauses.StringLiteral;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+public class MainTest extends BaseTest {
+    //Parsons with distractors
+    public Clause[] testSentence() {
+        return new Clause[]{
+                new StringLiteral("homecominghomecominghomecominghomecominghomecominghomecominghomecominghomecominghomecominghomecoming")
+        };
+    }
+
+    public void runMain() {
+        DupeDupeDupeDupe.main(new String[0]);
+    }
+
+    static Stream<Arguments> inputProvider() {
+        return Stream.of(
+                Arguments.of((Object) ArrayUtil.generateRandomWordArray(7)),
+                Arguments.of((Object) ArrayUtil.generateRandomWordArray(10)),
+                Arguments.of((Object) ArrayUtil.generateRandomWordArray(14)),
+                Arguments.of((Object) ArrayUtil.generateRandomWordArray(48)),
+                Arguments.of((Object) ArrayUtil.generateRandomWordArray(200)),
+                Arguments.of((Object) ArrayUtil.generateRandomWordArray(1000))
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    void correctStringRepeaterMethod(String[] in) throws Throwable {
+        Object[][] arguments = {
+                {in, String[].class}
+        };
+        MethodTest m = new MethodTest(DupeDupeDupeDupe.class, "stringRepeater", arguments);
+        Object output = m.callMethod();
+        CustomAssertions._assertEquals(getSolution(in), output, "Your stringRepeater method does not return a string with enough repetitions of the longest word.");
```

Comment:
Probably something about return a String with the correct number of repetitions of the longest word